src/Writers/HTMLWriter.jl

```diff
@@ -757,10 +757,12 @@ function render(doc::Documents.Document, settings::HTML=HTML())
         idx = findfirst(nn -> nn.page == page, doc.internal.navlist)
         nn = (idx === nothing) ? Documents.NavNode(page, nothing, nothing) : doc.internal.navlist[idx]
         @debug "Rendering $(page) [$(repr(idx))]"
-        render_page(ctx, nn)
+        render_page(doc, ctx, nn)
```

Comment:
You don't need to pass `doc`. You can always access it via `ctx.doc`.

Reply:
Oh i did not knew that i will change that.

Reply:
> Thanks for iterating on this! This exposes a completely new API for plugin writers, so it's worth taking some time to get this right.

I agree.

>     * I think it would be better to rely on multiple dispatch, rather than forcing the plugin writer to implement particular fields/properties. How about a function, like `extra_html_assets(::Plugin)`, that each plugin could implement, and would return the list of `HTMLAssets`?

Like those functions? They should do that only the names should be changed or are they not what you have in mind?
https://github.com/JuliaDocs/Documenter.jl/blob/77b7baf5911b3155332565e11020106afca49a8a/src/Writers/HTMLWriter.jl#L964-L983
Although the plugin struct fields could be optional. The moment fields with the same names would be used then there would it be necessary to check collisions/false input.

>     * Could you also add a new documentation page where we could tart documenting this API? I think it might be worth documenting it first to get the design down, and then focus on the implementation details.

A extra "Plug-in API" page Or a "How to Plug-in" page? I could start a page but i originally just wanted to add this and maybe refactor the asset addition structure for issue 1433. So i wont polish it, sorry.

TODO: I should also do asset input checking like here:
https://github.com/JuliaDocs/Documenter.jl/blob/77b7baf5911b3155332565e11020106afca49a8a/src/Writers/HTMLWriter.jl#L440-L444

Reply:
Changed it.